modules/aioseop_opengraph.php

```diff
@@ -1502,60 +1465,55 @@ function add_meta() {
 				// Set Twitter image from custom.
 				$twitter_thumbnail = set_url_scheme( $metabox['aioseop_opengraph_settings_customimg_twitter'] );
+			} elseif ( ! empty( $thumbnail ) ) {
+				$twitter_thumbnail = $thumbnail; // Default Twitter image if custom isn't set.
 			}
 
-			// Apply last filters.
-			$description = apply_filters( 'aioseop_description', $description );
-
+			/* *** COLLECT DATA *** */
 			$meta = array(
 				'facebook' => array(
-					'type'           => 'og:type',
-					'title'          => 'og:title',
-					'description'    => 'og:description',
-					'url'            => 'og:url',
-					'sitename'       => 'og:site_name',
-					'thumbnail'      => 'og:image',
-					'width'          => 'og:image:width',
-					'height'         => 'og:image:height',
-					'video'          => 'og:video',
-					'videowidth'     => 'og:video:width',
-					'videoheight'    => 'og:video:height',
-					'key'            => 'fb:admins',
-					'appid'          => 'fb:app_id',
-					'section'        => 'article:section',
-					'tag'            => 'article:tag',
-					'published_time' => 'article:published_time',
-					'modified_time'  => 'article:modified_time',
-					'publisher'      => 'article:publisher',
-					'author'         => 'article:author',
+					'og:type'                => $type,
+					'og:title'               => $title,
+					'og:description'         => $description,
+					'og:url'                 => $url,
+					'og:site_name'           => $sitename,
+					'og:image'               => $thumbnail,
+					'og:image:width'         => $width,
+					'og:image:height'        => $height,
+					'og:video'               => isset( $video ) ? $video : '',
+					'og:video:width'         => isset( $videowidth ) ? $videowidth : '',
+					'og:video:height'        => isset( $videoheight ) ? $videoheight : '',
+					'fb:admins'              => $key,
+					'fb:app_id'              => $appid,
+					'article:section'        => isset( $section ) ? $section : '',
+					'article:tag'            => $tag,
+					'article:published_time' => isset( $published_time ) ? $published_time : '',
+					'article:modified_time'  => isset( $modified_time ) ? $modified_time : '',
+					'article:publisher'      => isset( $publisher ) ? $publisher : '',
+					'article:author'         => isset( $author ) ? $author : '',
 				),
 				'twitter'  => array(
-					'card'              => 'twitter:card',
-					'site'              => 'twitter:site',
-					'creator'           => 'twitter:creator',
-					'domain'            => 'twitter:domain',
-					'title'             => 'twitter:title',
-					'description'       => 'twitter:description',
-					'twitter_thumbnail' => 'twitter:image',
+					'twitter:card'        => $card,
+					'twitter:site'        => $site,
+					'twitter:creator'     => $creator,
+					'twitter:domain'      => $domain,
+					'twitter:title'       => $title,
+					'twitter:description' => $description,
+					'twitter:image'       => $twitter_thumbnail,
```

Comment:
This code occurs three times. Wouldn't it be better to pull this into one or two separate functions?

Reply:
I believe you are referring to the reference array that occurs two times. It was intentionally done that way to make it painfully obvious to remove it when the filter hook is changed.

You are right though. I was split on the idea and you thought of it too, so I'll change it.